operator/src/main/java/oracle/kubernetes/operator/StuckPodProcessing.java

```diff
@@ -0,0 +1,132 @@
+  }
+
+  void checkStuckPods(String namespace) {
+    Step step = new CallBuilder()
+          .withLabelSelectors(LabelConstants.getCreatedbyOperatorSelector())
+          .listPodAsync(namespace, new PodListProcessing(namespace, SystemClock.now()));
+    mainDelegate.runSteps(step);
+  }
+
+  @SuppressWarnings("unchecked")
+  private List<V1Pod> getStuckPodList(Packet packet) {
+    return (List<V1Pod>) packet.computeIfAbsent("STUCK_PODS", k -> new ArrayList<>());
+  }
+
+  class PodListProcessing extends DefaultResponseStep<V1PodList> {
+
+    private final DateTime now;
+
+    public PodListProcessing(String namespace, DateTime dateTime) {
+      super(new PodActionsStep(namespace));
+      now = dateTime;
+    }
+
+    @Override
+    public NextAction onSuccess(Packet packet, CallResponse<V1PodList> callResponse) {
+      callResponse.getResult().getItems().stream()
+            .filter(pod -> isStuck(pod, now))
+            .forEach(pod -> addStuckPodToPacket(packet, pod));
+      
+      return doContinueListOrNext(callResponse, packet);
+    }
+
+    private boolean isStuck(V1Pod pod, DateTime now)  {
+      return getExpectedDeleteTime(pod).isBefore(now);
+    }
+
+    private DateTime getExpectedDeleteTime(V1Pod pod) {
+      return getDeletionTimeStamp(pod).plusSeconds((int) getDeletionGracePeriodSeconds(pod));
+    }
+
+    private long getDeletionGracePeriodSeconds(V1Pod pod) {
+      return Optional.of(pod).map(V1Pod::getMetadata).map(V1ObjectMeta::getDeletionGracePeriodSeconds).orElse(1L);
+    }
+
+    private DateTime getDeletionTimeStamp(V1Pod pod) {
+      return Optional.of(pod).map(V1Pod::getMetadata).map(V1ObjectMeta::getDeletionTimestamp).orElse(SystemClock.now());
+    }
+
+    private void addStuckPodToPacket(Packet packet, V1Pod stuckPod) {
+      getStuckPodList(packet).add(stuckPod);
+    }
+  }
+
+  class PodActionsStep extends Step {
+
+    private final String namespace;
+
+    public PodActionsStep(String namespace) {
+      this.namespace = namespace;
+    }
+
+    @Override
+    public NextAction apply(Packet packet) {
+      final List<V1Pod> stuckPodList = getStuckPodList(packet);
+      if (stuckPodList.isEmpty()) {
+        return doNext(packet);
+      } else {
+        Collection<StepAndPacket> startDetails = new ArrayList<>();
+
+        for (V1Pod pod : stuckPodList) {
+          startDetails.add(new StepAndPacket(createForcedDeletePodStep(pod), packet.clone()));
+        }
+        return doForkJoin(readExistingNamespaces(), packet, startDetails);
+      }
+    }
+
+    @Nonnull
+    private Step readExistingNamespaces() {
+      return mainDelegate.getDomainNamespaces().readExistingResources(namespace, mainDelegate.getDomainProcessor());
+    }
+
+    private Step createForcedDeletePodStep(V1Pod pod) {
+      return new CallBuilder()
+            .withGracePeriodSeconds(0)
+            .deletePodAsync(getName(pod), getNamespace(pod), getDomainUid(pod), null, new DefaultResponseStep<>());
```

Comment:
Does the DefaultResponseStep include typical retry behavior? We still have that issue where a delete will periodically fail because Kubernetes returns a V1Status rather than a V1Pod.

That's actually a little scary because this code might retry and accidentally interleave with make-right code triggered by the pod watcher that has already created a new pod.

Reply:
I was not aware of that possibility - that seems a bit scary for it to be non-determinate in its behavior.